Source/Shaders/ModelExperimental/ModelExperimentalVS.glsl

```diff
@@ -20,29 +20,50 @@ void main()
     dequantizationStage(attributes);
     #endif
 
-    // Update the position for this instance in place
-    #ifdef HAS_INSTANCING
-    instancingStage(attributes.positionMC);
-        #ifdef USE_PICKING
-        v_pickColor = a_pickColor;
-        #endif
-    #endif
 
     #if defined(HAS_FEATURES) && defined(FEATURE_ID_ATTRIBUTE)
     Feature feature;
     featureStage(feature);
     cpuStylingStage(attributes.positionMC, feature);
     updateFeatureStruct(feature);
     #endif
-    
+
+    mat4 modelView = czm_modelView;
+    mat3 normal = czm_normal;
+
+    // Update the position for this instance in place
+    #ifdef HAS_INSTANCING
+
+        // The legacy instance stage  is used when rendering I3DM models that 
+        // encode instances transforms in world space, as opposed to glTF models
+        // that use EXT_mesh_gpu_instancing, where instance transforms are encoded
+        // in object space.
+        #ifdef USE_LEGACY_INSTANCING
+        mat4 instanceModelView;
+        mat3 instanceModelViewInverseTranspose;
+        
+        legacyInstancingStage(attributes.positionMC, instanceModelView, instanceModelViewInverseTranspose);
+
+        modelView = instanceModelView;
+        normal = instanceModelViewInverseTranspose;
```

Comment:
Hm, could this be shortened to this?

```suggestion
        legacyInstancingStage(attributes.positionMC, modelView, normal);
```

Reply:
It could, but I deliberately left it like that to indicate that we do not use the `czm_modelView` or `czm_modelViewProjection` automatic uniforms, instead we compute separate ones just for the legacy instancing path. I like your cleaner approach too, but I want to make sure that this does not make it any less readable.